### PR TITLE
Fix backwards-incompatible calls to ASTHelpers.hasDirectAnnotationWithSimpleName

### PR DESCRIPTION
**Pull request by msridhar**
Error Prone 2.24.0 introduces new overloads of `ASTHelpers.hasDirectionAnnotationWithSimpleName`; see https://github.com/google/error-prone/commit/aa6e3e7be223e5f8319609281b229d4db4219654.  This causes backward compatibility issues when compiling against EP 2.24.0+ and then running on an older version of Error Prone.  We introduce a wrapper method to avoid this issue.  We also modify our `testJdk8` tasks to properly test the scenario of compiling against the newest supported Error Prone version and then running on the oldest supported version, which would have caught this issue.

Ideally we would have our own EP check preventing direct calls to `ASTHelpers.hasDirectionAnnotationWithSimpleName`, but that can be done in a follow-up.

### CODE REVIEW

File: nullaway/build.gradle
Comment (msridhar) [inline nullaway/build.gradle:24]: removed this configuration as it is no longer used

**Comment by msridhar** [comment]
@yuxincs this fixes the issue that caused release 0.10.20 to be unusable on older Error Prone versions

**Review by yuxincs — APPROVED** [review]
Nice fix! LGTM! I'll cut a new release once this and PR #895 land.

> Ideally we would have our own EP check

Agreed, let's handle that in follow-up PRs instead.

**Pull request merged**
